### DISCUSSION TITLE
How to build a system based on organizational units?

### DISCUSSION THREAD

**Discussion by 2op** [codeigniter4/shield#996] [Q&A]
I would like to build a system where users will be linked to an organization, for example:

- Organization 1
-- User 1
-- User 2
-- User X

- Organization 2
-- User 1
-- User 2
-- User X

Is it possible to do this with the shield?

**Comment by lonnieezell** [comment]
Hey @2op  - 

There's no way to do that out of the box. There's a couple of ways you could do that fairly easily, though: 

1) Add a new field to the `users` table, `organization_id`. Then you'd have to alter the registration process to know which company they were signing up through.
2) This actually would work in concert, but you could implement something similar to the Policies I implemented in our forum project (here's the [quick docs](https://github.com/lonnieezell/forum-example/blob/develop/_docs/policies.md) on it) to help with checking permissions within an org.

**Reply by 2op** [reply]
I understood! Thanks for the help @lonnieezell
